Framework/DataHandling/src/LoadAscii.cpp

```diff
@@ -199,11 +199,15 @@ API::Workspace_sptr LoadAscii::readData(std::ifstream &file) const {
     }
 
     for (size_t i = 0; i < numSpectra; ++i) {
-      spectra[i].dataX().push_back(values[0]);
-      spectra[i].dataY().push_back(values[i * 2 + 1]);
+      auto resizedHistogram = spectra[i].histogram();
+      const size_t newPosition = resizedHistogram.x().size();
+      resizedHistogram.resize(newPosition + 1);
+      resizedHistogram.mutableX()[newPosition] = values[0];
+      resizedHistogram.mutableY()[newPosition] = values[i * 2 + 1];
```

Comment:
There is also `back()` (and `front()`) for these types, just like for `std::vector`.